kernel/runtime/cxx/runtime/futex.cc

```diff
@@ -202,8 +283,8 @@ long do_futex(
         MLOG_WARN(mlog::app, "FUTEX_WAKE_BITSET");
         return futex_wake(uaddr, flags, val, val3);
     case FUTEX_REQUEUE:
-        MLOG_WARN(mlog::app, "FUTEX_REQUEUE");
-        return -ENOSYS;//futex_requeue(uaddr, flags, uaddr2, val, val2, NULL, 0);
+        //MLOG_DETAIL(mlog::app, "FUTEX_REQUEUE");
+        return futex_requeue(uaddr, flags, uaddr2, val, val2);
     case FUTEX_CMP_REQUEUE:
         MLOG_WARN(mlog::app, "FUTEX_CMP_REQUEUE");
         return -ENOSYS;//futex_requeue(uaddr, flags, uaddr2, val, val2, &val3, 0);
```

Comment:
Is it a lot of effort to also implement this? It seems like this was introduced to fix some bug in the mutex implementations, so there is a good chance we need that somewhen in the future.

Reply:
No, the difference to the implemented requeue function is a single check of the value pointed by uaddr. 
So, it is easy to be implemented when needed -.-